### ISSUE TITLE
Crossenv should support overriding CC, CXX, etc., as specified by host-python

### ISSUE THREAD

**Issue by virtuald** [benfogle/crossenv#16]
It seems that the environment doesn't get setup properly? I'm getting an error in my container:

```
root@8caac6e650ba:/build/venv/cross/bin# source activate
(cross) root@8caac6e650ba:/build/venv/cross/bin# python -m sysconfig
Platform: "l--x86_64"
Python version: "3.8"
Current installation scheme: "posix_prefix"

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.8/sysconfig.py", line 716, in <module>
    _main()
  File "/usr/lib/python3.8/sysconfig.py", line 710, in _main
    _print_dict('Paths', get_paths())
  File "/usr/lib/python3.8/sysconfig.py", line 502, in get_paths
    return _expand_vars(scheme, vars)
  File "/usr/lib/python3.8/sysconfig.py", line 172, in _expand_vars
    _extend_dict(vars, get_config_vars())
  File "/usr/lib/python3.8/sysconfig.py", line 550, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/usr/lib/python3.8/sysconfig.py", line 421, in _init_posix
    _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
ModuleNotFoundError: No module named '_sysconfigdata__arm-linux-gnueabi'
```

That file does exist in my cross python directory structure. Do you have any idea what might be missing? I'm happy to dig into it with a little direction...

**Comment by virtuald** [comment]
Actually, I was wrong, that file does not exist. However, site.py has the correct sysconfig name in it... for some reason it's failing? I'm guessing this is some new behavior that Python 3.8 introduced?

The file that exists is `robotpy/usr/local/lib/python3.8/_sysconfigdata__linux_arm-linux-gnueabi.py`, but by the time sysconfig gets executed, the return value of _get_sysconfigdata_name seems to be different?

**Comment by benfogle** [comment]
It should be possible. I think I need more information about how you are building Python. From the traceback it looks like you might be using the system version of Python. I'd generally recommend that you compile both the host and target versions of Python yourself, from the same source. The version that gets bundled with your distro often contains tweaks here and there that can cause unexpected incompatibilities.

**Comment by virtuald** [comment]
I've pushed the Dockerfile I'm using to setup the crossenv here: https://github.com/robotpy/robotpy-cross-docker .

I was using the deadsnakes python PPA to get Python 3.8 for Ubuntu 18, but I suppose I could compile it directly?

I compile the target (host) python directly from upstream source, no patches. You can see the configure argument here: https://github.com/robotpy/roborio-packages/blob/50d9029383d4f27b364780e62520f8ba3a16e793/ipkg/python3/Makefile#L5

**Comment by benfogle** [comment]
I've looked at this a little more, and the issue is definitely from using the deadsnakes Python.

The file that it's complaining about, `_sysconfigdata__arm-linux-gnueabi`, is determined by the function [`_get_sysconfigdata_name`](https://github.com/python/cpython/blob/3.8/Lib/sysconfig.py#L348) in the upstream source. The deadsnakes maintainers have [patched it](https://github.com/deadsnakes/python3.8/blob/ubuntu/bionic/debian/patches/0033-sysconfigdata-name.patch#L35) (with a commit message "Fix sysconfigdata module, more") such that it returns a different, incompatible name. There are likely to be many more little things like this, so you should definitely build your own version of build-python from the upstream source.

There's one more problem that I see, and I misunderstood a bit what you meant by natively compiled. Once I got it working, I noticed that the output of `python -m sysconfig` reports CC=`gcc -pthread` and CXX=`/usr/bin/g++ -pthread`. So when its time to cross compile, crossenv will simply use the version of gcc it finds on build machine, which is not what you want. Usually we expect CC to be something like `arm-linux-gnueabi-gcc` to avoid this issue.

So the answer to "Is it possible to use a natively complied python?" is no, not really, or at least, not at this time. I can suggest the following dirty hack in the meantime: Edit the file `/build/venv/cross/lib/_sysconfigdata__linux_arm-linux-gnueabi.py` and replace instances of `gcc` and `g++` with the appropriate path to your cross compiler.

**Comment by benfogle** [comment]
This would be useful for many prebuilt Pythons, where the cross compiler is often an absolute path that exists only in a Docker container. This would make testing easier as well.

**Comment by virtuald** [comment]
We got the cross-compilation environment working (thanks @DirtyJerz!). Looks like he had to setup the crossenv.cfg manually: https://github.com/robotpy/robotpy-cross-docker/pull/1/files

One weird thing that I got stuck on was that our `setup.py` has a build requirement ([for example..](https://github.com/robotpy/robotpy-wpiutil/blob/master/setup.py)), and so I kept trying to install it using `build-pip` as that's what was suggested in the readme. However, it kept failing. I ended up installing it using `pip` and then the wheels built fine.
